File: app/src/main/java/com/galtashma/parsedashboard/SortPreferenceStore.java
package com.galtashma.parsedashboard;

import com.appizona.yehiahd.fastsave.FastSave;

public class SortPreferenceStore {
    private String prefId;

    public SortPreferenceStore(String prefId) {
        this.prefId = prefId;
    }

    public void update(String key, boolean asc) {
        FastSave.getInstance().saveObject(this.prefId, new SortPreferenceItem(key, asc));
    }

    public String getKey() {
        if (isEmpty()) {
            return "";
        }
        return getSavedItem().key;
    }

    public boolean isAsc() {
        if (isEmpty()) {
            return false;
        }
        return getSavedItem().asc;
    }

    public boolean isEmpty() {
        return !FastSave.getInstance().isKeyExists(this.prefId);
    }

    private SortPreferenceItem getSavedItem() {
        return FastSave.getInstance().getObject(this.prefId, SortPreferenceItem.class);
    }

    class SortPreferenceItem {
        private String key;
        private boolean asc;

        SortPreferenceItem(String key, boolean asc) {
            this.key = key;
            this.asc = asc;
        }
    }
}

File: app/src/main/java/com/galtashma/parsedashboard/SortPreferenceStore.kt
package com.galtashma.parsedashboard

import com.appizona.yehiahd.fastsave.FastSave

/**
 * Created by gal on 3/16/18, rewritten by Cyb3rKo on 05/12/22.
 */

class SortPreferenceStore(private val prefId: String) {
    fun update(key: String, asc: Boolean) {
        FastSave.getInstance().saveObject(prefId, SortPreferenceItem(key, asc))
    }

    fun getKey() = if (!isEmpty()) getSavedItem().key else ""

    fun isAsc(): Boolean {
        return if (isEmpty()) {
            false
        } else {
            getSavedItem().asc
        }
    }

    fun isEmpty() = !FastSave.getInstance().isKeyExists(this.prefId)

    private fun getSavedItem(): SortPreferenceItem {
        return FastSave.getInstance().getObject(this.prefId, SortPreferenceItem::class.java)
    }

    class SortPreferenceItem(val key: String, val asc: Boolean)
}
